packages/wrangler/src/pages/dev.tsx

```diff
@@ -236,10 +232,42 @@ export const Handler = async ({
 		(promiseResolve) => (scriptReadyResolve = promiseResolve)
 	);
 
+	const workerScriptPath =
+		directory !== undefined
+			? join(directory, singleWorkerScriptPath)
+			: singleWorkerScriptPath;
+	const usingWorkerScript = existsSync(workerScriptPath);
+
+	const functionsDirectory = "./functions";
+	let usingFunctions = !usingWorkerScript && existsSync(functionsDirectory);
+
 	let scriptPath = "";
 
-	// Try to use Functions
-	if (usingFunctions) {
+	if (usingWorkerScript) {
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		scriptReadyResolve!();
+
+		scriptPath = workerScriptPath;
+
+		const runBuild = async () => {
+			try {
+				await esbuild.build({
+					entryPoints: [scriptPath],
+					write: false,
+					plugins: [blockWorkerJsImports],
+				});
+			} catch {}
+		};
+
+		await runBuild();
+		watch([scriptPath], {
```

Comment:
Does this watch only get cancelled if you kill the process?

Reply:
@petebacondarwin that is correct. In fact that is true for all our `watch`ers. I am rather new to `watch` and `esbuild` so my reasoning might be wrong, but the way I see it is that we want to keep watching these files (`_worker.js`, `/functions/*` and `_routes.json`) for changes for as long as the process is running, so we can serve the right assets. 
Is there a use case you had in mind that would require us to close the watcher? Or what was the underlying concern of your question?

Reply:
My only thought was whether esbuild kicks off a child process that might somehow get orphaned if we closed down the process in some other way than killing it? But I think that is probably not the case.

Reply:
according to https://github.com/evanw/esbuild/issues/643 this 👉 https://github.com/evanw/esbuild/blob/master/cmd/esbuild/service.go#L124-L137 should prevent something like that from happening. So I think we should be OK